### ISSUE TITLE
getState() does not handle cyclic object structures

### ISSUE THREAD

**Issue by floryst** [Kitware/vtk-js#620]
`getState()` from `macro.js` does not handle cyclic object structures.

Example: Calling `getState()` on an active `ViewProxy` with a proxyManager results in infinite recursion due to the chain `view.proxyManager.activeView.proxyManager.activeView...`, where `view` and `activeView` refer to the same view.

**Comment by jourdain** [comment]
Yes that is a known issue. If we solve that, you will need to fix the loading part handled by the `vtk()` function.

**Comment by jourdain** [comment]
In general, if you do a getState() on the renderWindow, you will get similar issue with the interactor...

**Comment by floryst** [comment]
Cyclic states can be fixed with something like https://www.npmjs.com/package/circular-json or https://github.com/WebReflection/flatted#flatted.

**Comment by jourdain** [comment]
Not really but that's ok. Solving that is not really difficult either.
